### PR TITLE
RELATED: RAIL-4112 Add support for filter drag and drop ordering

### PR DESCRIPTION
**Pull request by MilanLempera**
JIRA: RAIL-4112

---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8137/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9959/ : FAILURE in 4m 19s

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8148/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9970/ : FAILURE in 8m 16s

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8154/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9979/ : SUCCESS in 18m 54s

File: libs/sdk-ui-dashboard/.dependency-cruiser.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ options = {
         depCruiser.moduleWithDependencies("componentDefinition", "src/presentation/componentDefinition", [
             "src/model",
             "src/model/store/ui/uiSelectors.ts",
-            "src/presentation/filterBar",
+            "src/presentation/filterBar/*",
```
Comment (no23reason) [inline libs/sdk-ui-dashboard/.dependency-cruiser.js:19]: Why is the star needed? Maybe you are missing some export for the index file of the filterBar directory?

File: libs/sdk-ui-dashboard/src/presentation/filterBar/filterBar/DefaultFilterBar.tsx [hunk 2]
```diff
@@ -94,24 +95,28 @@ export function DefaultFilterBar(props: IFilterBarProps): JSX.Element {
                 {dateFilterMode === "hidden" ? (
```
Comment (no23reason) [inline libs/sdk-ui-dashboard/src/presentation/filterBar/filterBar/DefaultFilterBar.tsx:95]: The date filter should still be visible in edit mode, regardless of dateFilterMode. dateFilterMode should only be applied/respected in view mode.

File: libs/sdk-ui-dashboard/.dependency-cruiser.js (hunk 1 above)
Comment (MilanLempera) [reply libs/sdk-ui-dashboard/.dependency-cruiser.js:19]: I need to import filterBar/types. Re-export usage leads to a circular dependency.

File: libs/sdk-ui-dashboard/src/presentation/filterBar/filterBar/DefaultFilterBar.tsx (hunk 2 above)
Comment (MilanLempera) [reply libs/sdk-ui-dashboard/src/presentation/filterBar/filterBar/DefaultFilterBar.tsx:95]: fixed in selectEffectiveDateFilterMode

File: libs/sdk-ui-dashboard/.dependency-cruiser.js (hunk 1 above)
Comment (no23reason) [reply libs/sdk-ui-dashboard/.dependency-cruiser.js:19]: Makes sense, in that case please be explicit like here: https://github.com/gooddata/gooddata-ui-sdk/blob/master/libs/sdk-ui-dashboard/.dependency-cruiser.js#L133-L134

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8164/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9997/ : FAILURE in 6m 26s

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR2564

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/2336/ : SUCCESS in 2m 53s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2630/ : SUCCESS in 26s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1448/ : SUCCESS in 19m 10s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10001/ : SUCCESS in 26m 54s

**Pull request merged**